packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx

```diff
@@ -64,6 +72,50 @@ function MapboxMap({ onMapView, onPositionControl }) {
     }, []); // eslint-disable-line react-hooks/exhaustive-deps
     // We ignore eslint warnings about missing dependencies because onMapView should never change runtime and changing Mapbox Access Token runtime will give other problems.
 
+    // If the map is ready, check if the 3D features are toggled 
+    useEffect(() => {
+        if (mapView && mapView.isReady && isMapReady) {
+            // Check if the 3D features are switched on
+            // Conditionally render either 2D or 3D features depending on the value of is3DToggled
+            // Tilt the map to 45 degrees when the 3D features are switched on, and to 0 degrees when the 2D features are switched on.
+            if (is3DToggled) {
+                mapView.hideFeatures([mapView.MapboxFeatures.MODEL2D, mapView.MapboxFeatures.WALLS2D])
+
+                // If pitch is undefined, set the recoil value to 45 degrees and call the "tilt" method
+                if (pitch === undefined) {
+                    mapView.tilt(45, 2000);
+                    setPitch(45);
+                } else
+                // Else set the recoil value based on the pitch property and call the "tilt" method
+                {
+                    mapView.tilt(pitch, 2000);
+                    setPitch(pitch);
+                }
+            } else {
+                mapView.hideFeatures([mapView.MapboxFeatures.MODEL3D, mapView.MapboxFeatures.WALLS3D, mapView.MapboxFeatures.EXTRUSION3D, mapView.MapboxFeatures.EXTRUDEDBUILDINGS])
+                mapView.tilt(0, 2000);
+            }
+        }
+    }, [mapView, is3DToggled, isMapReady, mapView?.isReady, pitch]);
+
+    /*
+     * React on changes to the solution.
+     * Decide whether to show the VisibilitySwitch component depending on the modules enabled on the solution.
+     */
+    useEffect(() => {
+        if (solution) {
+            const isMapboxModuleEnabled = solution.modules.map(module => module.toLowerCase()).includes('mapbox');
+            const is3DWallsModuleEnabled = solution.modules.map(module => module.toLowerCase()).includes('3dwalls');
+            // The module floorplan refers to 2D Walls 
+            const is2DWallsModuleEnabled = solution.modules.map(module => module.toLowerCase()).includes('floorplan');
+
+            if (isMapboxModuleEnabled && is3DWallsModuleEnabled && is2DWallsModuleEnabled) {
+                setShowVisibilitySwitch(true);
+            }
```

Comment:
Can be shortened to 

```js
            // If all of the modules "mapbox", "3dwalls" and "floorplan" (2D walls) are enabled on the solution, show the VisibilitySwitch component
            if (['mapbox', '3dwalls', 'floorplan'].every(requiredMoule => solution.modules.map(module => module.toLowerCase()).includes(requiredMoule))) {
                setShowVisibilitySwitch(true);
            }
```
but I'm not sure if that is more readable.

Reply:
Yes, it would be! I made the change ✅